File: Golang/SearchInsertPosition/SearchInsertPosition_test.go
package SearchInsertPosition

import "testing"

func TestSearchInsert(t *testing.T) {

	if searchInsert([]int{1}, 0) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if searchInsert([]int{1, 3, 5, 6}, 5) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if searchInsert([]int{1, 3, 5, 6}, 2) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if searchInsert([]int{1, 3, 5, 6}, 7) == 4 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if searchInsert([]int{1, 3, 5, 6}, 0) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/SearchInsertPositionTest.py
from unittest import TestCase

from SearchInsertPosition import SearchInsertPosition


class TestSearchInsertPosition(TestCase):
    def test_searchInsert(self):
        sip = SearchInsertPosition()

        self.assertTrue(sip.searchInsert([1], 0) == 0)

        self.assertTrue(sip.searchInsert([1], 2) == 1)

        self.assertTrue(sip.searchInsert([1, 3, 5, 6], 5) == 2)

        self.assertTrue(sip.searchInsert([1, 3, 5, 6], 2) == 1)

        self.assertTrue(sip.searchInsert([1, 3, 5, 6], 7) == 4)

        self.assertTrue(sip.searchInsert([1, 3, 5, 6], 0) == 0)
